### ISSUE TITLE
Issues are not labeled

### ISSUE THREAD

**Issue by fabacino** [abo-abo/swiper#1123]
Sometimes I browse through the open issues searching for something I can help with. However, since they are not labeled, I basically have to go through the issues one by one to find out if it is a bug, a feature request or something else like a question which has already been answered but not confirmed by the opener yet. This process is quite tedious and time consuming.

Therefore, I would like to ask if we could start assigning labels to the issues. In addition to the standard ones I would suggest another one called `Awaiting feedback` or something like that which could be assigned to issues where a response has been given to the opener but there is no feedback yet.

I think labels would have a positive effect on the number of open issues as it would be immediately clear which issues actually require work and which ones are just waiting for feedback or are not going to be fixed. I am sure it would be of great healp to people who would like to contribute and also provide a means to get a better overview (e.g. of the 131 open issues, 16 are bugs, 64 are feature requests, 32 are awaiting feedback, ...).

If you think this is a good idea, I am willing to help you with the labeling.

**Comment by abo-abo** [comment]
> If you think this is a good idea, I am willing to help you with the labeling.

I think it's a great idea. I want to welcome all the help I can get on the project.

Unfortunately, I don't have as much free time as I used to have due to a job change. Regarding the growing amount of issues, I had the following idea:

1. Make a template org-mode document inside the repository that explains what a reproducible issue is, and how to do it.
2. Make a template org-mode document inside the repository about issue triage that says something like "Please react within 7 days to confirm that the issue is still reproducible or the issue will be closed automatically. You can still open a new issue with the same topic if it's still relevant. In that case, please extract all the relevant info from the old closed issue into the new one, so that the person fixing
it doesn't have to dig through threads.". This template will reference item 1.
3. Write a copy-pasted message with nothing but the reference to the template from item 2 into each of 130 issues. If there's no reaction in 7 days, close the issue.
4. Make an org-mode document inside the repo for feature requests. I think if a feature request is around for more than 1 month, it's probably hard to resolve. Migrate it out of the way into this document. It becomes lower priority, like a someday-maybe thing, but it's not forgotten and it doesn't get in the way any more.

Let me know what you think about these ideas. Maybe you could help with them as well. Or you could just do issue labeling, whatever you prefer.

**Comment by basil-conto** [comment]
> You can still open a new issue with the same topic if it's still relevant. In that case, please extract all the relevant info from the old closed issue into the new one, so that the person fixing
it doesn't have to dig through threads.

I'm just curious - is this any different/better than reopening the expired GitHub issue?

**Comment by abo-abo** [comment]
> I'm just curious - is this any different/better than reopening the expired GitHub issue?

I think it's better. You don't have to dig through potentially months of discussion just to understand what the issue is about. Instead, a nice short summary of the things that are still relevant.

**Comment by basil-conto** [comment]
Makes sense, thanks.

**Comment by fabacino** [comment]
> 1. Make a template org-mode document inside the repository that explains what a reproducible issue is, and how to do it.

Good idea. In addition, I would add a paragraph to the "Contributing" section of the README file with a link to this template as well.

> 2. Make a template org-mode document inside the repository about issue triage that says something like "Please react within 7 days to confirm that the issue is still reproducible or the issue will be closed automatically. You can still open a new issue with the same topic if it's still relevant. In that case, please extract all the relevant info from the old closed issue into the new one, so that the person fixing
    it doesn't have to dig through threads.". This template will reference item 1.
>
> 3. Write a copy-pasted message with nothing but the reference to the template from item 2 into each of 130 issues. If there's no reaction in 7 days, close the issue.

Sounds good and should bring down the number of open issues quite a bit. After those 7 days, we can start labeling the remaining issues.

> 4. Make an org-mode document inside the repo for feature requests. I think if a feature request is around for more than 1 month, it's probably hard to resolve. Migrate it out of the way into this document. It becomes lower priority, like a someday-maybe thing, but it's not forgotten and it doesn't get in the way any more.

I see a couple of problems with putting older feature requests aside in a file:

1. There are two places to look for the same thing, which usually is something to avoid.

2. As a consequence of 1, feature requests listed in the file will have a harder time finding somebody to implement them, since most users will probably look through the open issues list first before consulting the file, even though the file might contain feature requests which would be of more value to the end-user.

3. Users might be more reluctant to participate in discussions because the issue is "out of sight". Let's say there is an issue about a new feature for ivy but since nobody had time to implement it the issue gets closed and the feature request is moved into the file. Now after some time a user might have new information about it, maybe the feature doesn't make sense anymore or you can just do X with emacs version Y or install package Z to get the same effect. If he finds the issue in question already closed however, he might not share that information because he might think that it's of no use anymore as the issue is already done.
Sure you can mention the org-file in a comment or add a special label, but I think my first reaction as a user not very familiar with the project would still be "the issue is already closed, so there's nothing going on here anymore".

I see your point about feature requests lying around forever and keeping the issue count high but I would still suggest to keep all of them open regardless of their age, at least at first. After all, they don't get less important just because they are older. We can just assign appropriate labels to those features who are low priority or hard to implement, so they can be filtered out if wanted.

I think the main problem with the issues at the moment is not that there are too many but that there are too many with an unknown status. Even leaving all the feature requests open, I am sure you can easily filter or search through the issues once all the remaining ones are labeled properly. If not and those older issues turn out to get in the way more than (I) expected, we can still do the moving into the file in a second step.
